src/emulator/auth/handlers.ts

```diff
@@ -249,28 +249,26 @@ export function registerHandlers(
       return { status: 'ACK', webStorageSupport: true };
     }, gapi.iframes.CROSS_ORIGIN_IFRAMES_FILTER);
 
+    var authEvent = null;
     var storedEvent = sessionStorage['firebase:redirectEvent:' + storageKey];
     if (storedEvent) {
-      var authEvent = null;
       try {
         authEvent = JSON.parse(storedEvent);
       } catch (_) {
         return alert('Auth Emulator Internal Error: Invalid stored event.');
       }
-      if (authEvent) {
-        sendAuthEvent(parentContainer, authEvent);
-      }
-      delete sessionStorage['firebase:redirectEvent:' + storageKey];
     }
+    sendAuthEvent(authEvent);
+    delete sessionStorage['firebase:redirectEvent:' + storageKey];
   }
 
-  function sendAuthEvent(parentContainer, authEvent) {
+  function sendAuthEvent(authEvent) {
     parentContainer.send('authEvent', {
-      type: 'authEvent',
-      authEvent: authEvent,
+      type: authEvent ? 'authEvent' : 'unknown',
+      authEvent: authEvent || {},
     }, function(responses) {
       if (!responses || !responses.length ||
-          !responses[responses.length - 1].status === 'ACK') {
+          responses[responses.length - 1].status !== 'ACK') {
         return alert("Auth Emulator Internal Error: Sending authEvent failed.");
```

Comment:
This throws an alert up whenever the iframe is initialized

Reply:
Reverted for now. This condition is really messed up and misleading, and I'll look into it later.

Reply:
Okay cool let me play with it tomorrow but otherwise lgtm :+1:

Reply:
This works in my testing, thanks again!

Reply:
Fix for the condition: https://github.com/firebase/firebase-tools/pull/3188